From 1094b4f3149cde8ea3b80b46d62f59a6d1cfae97 Mon Sep 17 00:00:00 2001
From: DosWorld <45263847+DosWorld@users.noreply.github.com>
Date: Fri, 29 Oct 2021 09:12:45 +0300
Subject: [PATCH] Fix broken RDF2 output

Cause "rdf segment numbers not allocated as expected (2,4,6)"

See outrdf2.c:

    segtext = seg_alloc();
    segdata = seg_alloc();
    segbss = seg_alloc();
    if (segtext != 0 || segdata != 2 || segbss != 4)
        nasm_panic("rdf segment numbers not allocated as expected (%d,%d,%d)",
                   segtext, segdata, segbss);
---
 asm/segalloc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/asm/segalloc.c b/asm/segalloc.c
index 6d7a4202..4f56cd7d 100644
--- a/asm/segalloc.c
+++ b/asm/segalloc.c
@@ -40,7 +40,7 @@
 #include "nasmlib.h"
 #include "insns.h"
 
-static int32_t next_seg  = 2;
+static int32_t next_seg  = 0;
 
 int32_t seg_alloc(void)
 {
